src/types/type_row.rs

```diff
@@ -39,36 +39,29 @@ impl TypeRow {
         }
     }
 
-    /// Iterator over the types in the row.
-    pub fn iter(&self) -> impl Iterator<Item = &Type> {
-        self.types.iter()
-    }
+    delegate! {
+        to self.types {
```

Comment:
I forgot that docs are not inherited 😓 
Maybe I'll make an issue for that in delegate's repo.

You can drop the #inlines, they are automatically added.